### PR TITLE
Use try-with-resources Statement and avoid closing it manually.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: impl/src/main/java/org/apache/myfaces/config/impl/FacesConfigUnmarshallerImpl.java [hunk 1]
```diff
@@ -179,18 +180,8 @@ public FacesConfigImpl getFacesConfig(InputStream in, String systemId) throws IO
...
-            {
-                in.close();
-            }
-            catch (IOException e)
-            {
-                // ignore silently
-            }
-        }
-          
+        // ignore silently
```
Comment (melloware) [inline impl/src/main/java/org/apache/myfaces/config/impl/FacesConfigUnmarshallerImpl.java:183]: I don't know if you need these trailing comments?

File: impl/src/main/java/org/apache/myfaces/resource/TempDirFileCacheContractResourceLoader.java [hunk 2]
```diff
@@ -229,20 +227,7 @@ protected void createTemporalFileVersion(FacesContext facesContext, ResourceMeta
...
-                {
-                    inputStream.close();
-                }
-                catch (IOException e)
-                {
-                    // Ignore
-                }
-            }
-        }
+        // Ignore
```
Comment (melloware) [inline impl/src/main/java/org/apache/myfaces/resource/TempDirFileCacheContractResourceLoader.java:230]: Same?

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by tandraschko** [comment]
also it would throw a IOException when close fails? currently we ignore it

File: impl/src/main/java/org/apache/myfaces/config/impl/FacesConfigUnmarshallerImpl.java (hunk 1 above)
Comment (arturobernalg) [reply impl/src/main/java/org/apache/myfaces/config/impl/FacesConfigUnmarshallerImpl.java]: done

File: impl/src/main/java/org/apache/myfaces/resource/TempDirFileCacheContractResourceLoader.java (hunk 2 above)
Comment (arturobernalg) [reply impl/src/main/java/org/apache/myfaces/resource/TempDirFileCacheContractResourceLoader.java]: done

**Comment by arturobernalg** [comment]
> also it would throw a IOException when close fails? currently we ignore it

Well, changed. but in the class TempDirFileCacheContractResourceLoader its not possible without break the compatibility.

**Comment by tandraschko** [comment]
We dont use final
Also we should not throw a facesexception

**Comment by tandraschko** [comment]
I think we should not touch all cases where close ist silently ignore

**Comment by arturobernalg** [comment]
> I think we should not touch all cases where close ist silently ignore

Now I'm lost.

**Comment by tandraschko** [comment]
there is no nice way to silnently ignore ioexception only on #close in try-resource
therefore i just pushed: https://github.com/apache/myfaces/commit/b79b08ece301f2959332df0b950c4bcc712f360e

**Pull request closed without merging**